atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterTests.java

```diff
@@ -0,0 +1,467 @@
+
+    private Set<String> listAllTablesWithMapping() {
+        return sqliteOracleAdapter.listAllTablesWithMapping();
+    }
+
+    private Set<TableReference> listAllTablesWithMetadata() {
+        return sqliteOracleAdapter.listAllTablesWithMetadata();
+    }
+
+    private NamespaceDeleter createDefaultNamespaceDeleter() {
+        return new OracleNamespaceDeleter(
+                createDefaultNamespaceDeleterParameters().build());
+    }
+
+    private NamespaceDeleter createNonMappingNamespaceDeleter() {
+        return new OracleNamespaceDeleter(createNamespaceDeleterParameters(
+                        getDefaultDdlConfig().useTableMapping(false).build())
+                .build());
+    }
+
+    private ImmutableOracleDdlConfig.Builder getDefaultDdlConfig() {
+        return ImmutableOracleDdlConfig.builder()
+                .tablePrefix(TABLE_PREFIX)
+                .overflowTablePrefix(OVERFLOW_TABLE_PREFIX)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+                .useTableMapping(true)
+                .metadataTable(SqliteOracleAdapter.METADATA_TABLE);
+    }
+
+    private ImmutableOracleNamespaceDeleterParameters.Builder createDefaultNamespaceDeleterParameters() {
+        return createNamespaceDeleterParameters(getDefaultDdlConfig().build());
+    }
+
+    private ImmutableOracleNamespaceDeleterParameters.Builder createNamespaceDeleterParameters(
+            OracleDdlConfig ddlConfig) {
+        OracleTableNameGetter tableNameGetter = OracleTableNameGetterImpl.createDefault(ddlConfig);
+        ConnectionSupplier connectionSupplier =
+                new ConnectionSupplier(sqliteOracleAdapter.createSqlConnectionSupplier());
+        Function<TableReference, OracleDdlTable> ddlTableFactory = tableReference -> OracleDdlTable.create(
+                tableReference,
+                connectionSupplier,
+                ddlConfig,
+                tableNameGetter,
+                new TableValueStyleCache(),
+                executorService);
+        return OracleNamespaceDeleterParameters.builder()
+                .tablePrefix(ddlConfig.tablePrefix())
+                .overflowTablePrefix(ddlConfig.overflowTablePrefix())
+                .userId(TEST_USER)
+                .oracleDdlTableFactory(ddlTableFactory)
+                .connectionSupplier(connectionSupplier)
+                .tableNameGetter(tableNameGetter);
+    }
+
+    private OracleNamespaceDeleterParameters createNamespaceDeleterParametersWithUnstableFactory() {
+        OracleNamespaceDeleterParameters parameters =
+                createDefaultNamespaceDeleterParameters().build();
+        return OracleNamespaceDeleterParameters.builder()
+                .from(parameters)
+                .oracleDdlTableFactory(new UnstableOracleDdlFactory(parameters.oracleDdlTableFactory()))
+                .build();
+    }
+
+    private static final class UnstableOracleDdlFactory implements Function<TableReference, OracleDdlTable> {
```

Comment:
nit: docs, or rename to maybe `OnceFailingOracleDdlFactory`? Main thing is I perceive `Unstable` to fail infinitely often as infinitely many requests are being made, and while that's _technically_ true, 1 in 2^32 is not really unstable.